lib/internal/http2/compat.js

```diff
@@ -649,17 +654,28 @@ class Http2ServerResponse extends Stream {
   }
 
   write(chunk, encoding, cb) {
+    const state = this[kState];
+
     if (typeof encoding === 'function') {
       cb = encoding;
       encoding = 'utf8';
     }
 
-    if (this[kState].closed) {
-      const err = new ERR_HTTP2_INVALID_STREAM();
-      if (typeof cb === 'function')
-        process.nextTick(cb, err);
-      else
-        throw err;
+    if (typeof cb !== 'function')
+      cb = nop;
+
+    let err;
+    if (state.ending) {
+      err = new ERR_STREAM_WRITE_AFTER_END();
+    } else if (state.destroyed) {
+      err = new ERR_STREAM_DESTROYED('write');
```

Comment:
This is explicitly different from `http` which fails silently when the socket is destroyed but we're not aware of it actually being closed. That presents a problem if we think about it from a compatibility perspective.

Reply:
To expand on this: right now http1 will only error with `ERR_STREAM_WRITE_AFTER_END` if you explicitly call `end`. In all other cases it just fails silently. We probably also need to remove the `state.closed` error to get closer to compatibility.